chia/rpc/wallet_rpc_api.py

```diff
@@ -924,6 +925,17 @@ async def create_offer_for_ids(self, request):
         offer: Dict[str, int] = request["offer"]
         fee: uint64 = uint64(request.get("fee", 0))
         validate_only: bool = request.get("validate_only", False)
+        type_dict_str: Optional[Dict[str, str]] = request.get("type_dict", None)
+
+        # This type_dict construction is to maintain backward compatibility where everything is assumed to be a CAT
+        type_dict: Dict[str, str] = {}
+        if type_dict_str is None:
+            for key in offer:
+                if len(key) >= 32:
+                    type_dict[bytes32.from_hexstr(key)] = AssetType.CAT
```

Comment:
`bytes32`'s initializer requires exactly 32 bytes, otherwise it will raise a `ValueError`. So I think the len check should be `if len(key) == 64:`